homeassistant/components/google_assistant/trait.py

```diff
@@ -566,15 +586,24 @@ def query_attributes(self):
         if current_humidity is not None:
             response['thermostatHumidityAmbient'] = current_humidity
 
-        if (operation == climate.STATE_AUTO and
-                climate.ATTR_TARGET_TEMP_HIGH in attrs and
-                climate.ATTR_TARGET_TEMP_LOW in attrs):
-            response['thermostatTemperatureSetpointHigh'] = \
-                round(temp_util.convert(attrs[climate.ATTR_TARGET_TEMP_HIGH],
-                                        unit, TEMP_CELSIUS), 1)
-            response['thermostatTemperatureSetpointLow'] = \
-                round(temp_util.convert(attrs[climate.ATTR_TARGET_TEMP_LOW],
-                                        unit, TEMP_CELSIUS), 1)
+        if operation == climate.STATE_AUTO:
```

Comment:
This change will break a lot of things. We map state auto to heat/cool in GA as that is the only mode that supports setting high and low temperatures.

Reply:
This change only has an impact if:

* The device itself has an auto mode
* In auto mode the device doesn't accept a range, but only a single value

If these conditions occur, GA just doesn't report a temperature at all. If you return a range with high and low set to the same value this gets displayed as a single value in GA and you can update it in GA, but when it sends an update it sends a range. Sending this downstream to the device again doesn't work because the device doesn't accept a range.

Fixing this in the climate component is the wrong place to fix it, if the device only has a single value rather than a range, then this displays correctly in HA but doesn't work through in GA.

So, in summary, the patch won't break anything that isn't already broken.

Reply:
For sure it *should*, but the device I have doesn't do this, it only has a single value. We can't force the rest of the world into our mapping of it. If the device doesn't (and sure maybe it should, but it doesn't) then this change supports that the best.

Reply:
We control the state mapping for the components, the component should be fixed to reflect it's actual state. In HA, the powers that be told me when I suggested something like this a few months ago said that the "auto" mode is for dual set points. If a device only supports one set point, it should report heat or cool.

Reply:
But what if the auto mode doesn't have a set range, but a single value? It doesn't make a lot of sense to report a range when there's only a single value. You'd end up having a range displayed in HA, but the cool and heat targets would just be linked together causing needless UI clutter.

Reply:
I've tested this change in GA of course, the slider displays the setpoint and you can change only the single value, it just sends a range when you take your finger off the slider.

Reply:
The tl:dr is, if it only accepts one set point, don't use auto

Reply:
Hey, can I ask you to join the community discord? The back and forth on comments is getting messy.